libparsec/crates/platform_mountpoint/src/unix/mod.rs

```diff
@@ -0,0 +1,380 @@
+
+        match self.file_delete(&path) {
+            Ok(_) => reply.ok(),
+            Err(_) => reply.error(ENOENT),
+        }
+    }
+
+    fn mkdir(
+        &mut self,
+        _req: &Request<'_>,
+        parent: u64,
+        name: &OsStr,
+        _mode: u32,
+        _umask: u32,
+        reply: ReplyEntry,
+    ) {
+        let parent_path = self.get_path(parent.into());
+        let path = parent_path.join(name);
+
+        match self.dir_create(&path) {
+            Ok(_) => {
+                let entry = self.entry_info(&path).expect("Path just created");
+                let ino = self.insert_path(path);
+                reply.entry(&TTL, &entry.to_file_attr(ino), GENERATION)
+            }
+            Err(MountpointError::NameTooLong) => reply.error(ENAMETOOLONG),
+            Err(_) => reply.error(ENOENT),
+        }
+    }
+
+    fn rmdir(&mut self, _req: &Request, parent: u64, name: &OsStr, reply: ReplyEmpty) {
+        let parent_path = self.get_path(parent.into());
+        let path = parent_path.join(name);
+
+        match self.dir_delete(&path) {
+            Ok(_) => reply.ok(),
+            Err(MountpointError::DirNotEmpty) => reply.error(ENOTEMPTY),
+            Err(_) => reply.error(ENOENT),
+        }
+    }
+
+    fn rename(
+        &mut self,
+        _req: &Request<'_>,
+        parent: u64,
+        name: &OsStr,
+        newparent: u64,
+        newname: &OsStr,
+        _flags: u32,
+        reply: ReplyEmpty,
+    ) {
+        let parent_path = self.get_path(parent.into());
+        let source = parent_path.join(name);
+
+        let newparent_path = self.get_path(newparent.into());
+        let destination = newparent_path.join(newname);
+
+        match self.entry_rename(&source, &destination, true) {
+            Ok(_) => {
+                self.rename_path(&source, &destination);
+                reply.ok()
+            }
+            Err(_) => reply.error(ENOENT),
+        }
+    }
+
+    fn flush(&mut self, _req: &Request, _ino: u64, fh: u64, _lock_owner: u64, reply: ReplyEmpty) {
+        self.fd_flush(FileDescriptor(fh as u32));
+        reply.ok();
+    }
+
+    fn fsync(&mut self, _req: &Request, _ino: u64, fh: u64, _datasync: bool, reply: ReplyEmpty) {
+        self.fd_flush(FileDescriptor(fh as u32));
+        reply.ok();
+    }
+
+    fn fsyncdir(
+        &mut self,
+        _req: &Request,
+        _ino: u64,
+        _fh: u64,
+        _datasync: bool,
+        reply: ReplyEmpty,
+    ) {
+        // todo
+        reply.ok();
+    }
+}
+
+pub struct FileSystem {
+    unmounter: SessionUnmounter,
+    mountpoint: PathBuf,
+}
+
+// We do the same as WinFSP: remove the directory
+impl FileSystem {
+    pub fn stop(mut self) {
+        let _ = self.unmounter.unmount();
+        let _ = std::fs::remove_dir(self.mountpoint);
+    }
+}
+
+// We do the same as WinFSP: try to create a directory
+pub fn mount(mountpoint: &Path) -> anyhow::Result<FileSystem> {
```

Comment:
This function should get a `impl MountpointInterface` as parameter, otherwise it is not possible to use it with the actual parsec mountpoint !

Reply:
What do you mean ?

Reply:
Done